src/mlpack/methods/reinforcement_learning/environment/continuous_mountain_car.hpp

```diff
@@ -0,0 +1,194 @@
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 2;
+
+   private:
+    //! Locally-stored velocity and position vector.
+    arma::colvec data;
+  };
+
+  /**
+   * Implementation of action of Continuous Mountain Car.
+   */
+  struct Action
+  {
+    double action[1];
+    // Storing degree of freedom
+    const int size = 1;
+  };
+
+  /**
+   * Construct a Continuous Mountain Car instance using the given constant.
+   *
+   * @param positionMin Minimum legal position.
+   * @param positionMax Maximum legal position.
+   * @param velocityMin Minimum legal velocity.
+   * @param velocityMax Maximum legal velocity.
+   */
+  ContinuousMountainCar(const double positionMin = -1.2,
+                        const double positionMax = 0.5,
+                        const double velocityMin = -0.07,
+                        const double velocityMax = 0.07) :
+      positionMin(positionMin),
+      positionMax(positionMax),
+      velocityMin(velocityMin),
+      velocityMax(velocityMax)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Continuous Mountain Car. Get reward and next state based 
+   * on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, it's always -1.0.
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState) const
+  {
+    // Calculate acceleration.
+    double force = std::min(std::max(action.action[0], -1.0), 1.0);
+    nextState.Velocity() = state.Velocity() + 0.001 * force - 0.0025 *
+        std::cos(3 * state.Position());
+    nextState.Velocity() = std::min(
+        std::max(nextState.Velocity(), velocityMin), velocityMax);
+
+    // Update states.
+    nextState.Position() = state.Position() + nextState.Velocity();
+    nextState.Position() = std::min(
+        std::max(nextState.Position(), positionMin), positionMax);
+
+    if (nextState.Position() == positionMin && nextState.Velocity() < 0)
+    {
+      nextState.Velocity() = 0.0;
+    }
+    // If it is a terminal state, reward is 100.0
+    if (IsTerminal(nextState))
+      return 100.0;
+    return -pow(action.action[0], 2)*0.1;
+  }
+
+  /**
+   * Dynamics of Continuous Mountain Car. Get reward based on current state and current
+   * action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @return reward, it's always -1.0.
+   */
+  double Sample(const State& state, const Action& action) const
+  {
+    State nextState;
+    return Sample(state, action, nextState);
+  }
+
+  /**
+   * Initial position is randomly generated within [-0.6, -0.4].
+   * Initial velocity is 0.
+   *
+   * @return Initial state for each episode.
+   */
+  State InitialSample() const
+  {
+    State state;
+    state.Velocity() = 0.0;
+    state.Position() = arma::as_scalar(arma::randu(1)) * 0.2 - 0.6;
```

Comment:
What about using:

https://github.com/mlpack/mlpack/blob/114137eec8c025d62e77e2c9379ce5f0f76943b9/src/mlpack/core/math/random.hpp#L71-L74

that way we could avoid ``arma::as_scalar``.